### ISSUE TITLE
LSP completions include unexpected symbols

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#41841]
Give this code:

```dart
void main() {
  '${fo}';
  Env
}

void foo() {}
```

Invoking completion after `Env` on line 3 results in failures to resolve any of the completion items:

```
[11:45:14 AM] [Analyzer] [Info] ==> {"jsonrpc":"2.0","id":39,"method":"completionItem/resolve","params":{"label":"environment","detail":"Map<String, String>","insertTextFormat":1,"kind":10,"sortText":"999997","data":{"file":"/Users/danny/Desktop/dart_sample/bin/string_interp.dart","offset":30,"libId":136,"displayUri":"dart:io","rOffset":27,"rLength":3}}}
[11:45:14 AM] [Analyzer] [Info] <== Content-Length: 111
Content-Type: application/vscode-jsonrpc; charset=utf-8
[11:45:14 AM] [Analyzer] [Info] <== {"id":39,"jsonrpc":"2.0","error":{"code":-32602,"message":"No such element: environment","data":"environment"}}
[11:45:15 AM] [Analyzer] [Info] ==> Content-Length: 309
[11:45:15 AM] [Analyzer] [Info] ==> {"jsonrpc":"2.0","id":40,"method":"completionItem/resolve","params":{"label":"endValidity","detail":"DateTime","insertTextFormat":1,"kind":10,"sortText":"999997","data":{"file":"/Users/danny/Desktop/dart_sample/bin/string_interp.dart","offset":30,"libId":136,"displayUri":"dart:io","rOffset":27,"rLength":3}}}
[11:45:15 AM] [Analyzer] [Info] <== Content-Length: 111
Content-Type: application/vscode-jsonrpc; charset=utf-8
[11:45:15 AM] [Analyzer] [Info] <== {"id":40,"jsonrpc":"2.0","error":{"code":-32602,"message":"No such element: endValidity","data":"endValidity"}}
```

Using Dart nightly from a few days ago (2.9.0-edge.3788a59c6cd72378fe9cc139829cafe40c819079).

**Comment by DanTup** [comment]
Note: I had auto-import-completions in VS Code disabled when this happened.

**Comment by DanTup** [comment]
The symbols that are failing look suspicious - I don't think endValidity should be showing up here:

<img width="575" alt="Screenshot 2020-05-11 at 12 16 56" src="https://user-images.githubusercontent.com/1078012/81555956-630bbf80-9381-11ea-883e-ee59ef3ff18b.png">

This results in:

[Error - 12:16:00 PM] Request completionItem/resolve failed.
  Message: No such element: endValidity in dart:io
  Code: -32602 

Coming from:

https://github.com/dart-lang/sdk/blob/91d6e1a0a712a308521893784d858b322bb59e86/pkg/analysis_server/lib/src/lsp/handlers/handler_completion_resolve.dart#L103-L111

**Comment by DanTup** [comment]
Got a fix at https://dart-review.googlesource.com/c/sdk/+/147545/.

**Comment by DanTup** [comment]
Fixed in 9678d604c7c70a87402ce82f5c7db05c573e0ed6.
